### PR TITLE
(CDAP-7237) Appended principal to dot role name to avoid conflicts

### PR DESCRIPTION
**Pull request by rohitsinha54**
Issue: https://issues.cask.co/browse/CDAP-7237
Creating PR on from @bdmogal branch.

### CODE REVIEW

File: cdap-sentry/cdap-sentry-extension/cdap-sentry-binding/src/main/java/co/cask/cdap/security/authorization/sentry/binding/SentryAuthorizer.java [hunk 1]
```diff
@@ -182,4 +183,23 @@ private String getRequestingUser() throws IllegalArgumentException {
...
+      case GROUP:
+        principalType = "g";
+        break;
+      case ROLE:
+        principalType = "r";
+        break;
+      default:
+        throw new IllegalStateException("Invalid principal type " + principal.getType());
+    }
+    return Joiner.on(ENTITY_ROLE_PREFIX).join("", entityId.toString(), principal.getName(), principalType);
```
Comment (rohitsinha54) [inline cdap-sentry/cdap-sentry-extension/cdap-sentry-binding/src/main/java/co/cask/cdap/security/authorization/sentry/binding/SentryAuthorizer.java]: Why do we need "" ?
Also, get we just get the first character from .name().toLowerCase().charAt(0) and get rid of the switch case.

Comment (bdmogal) [reply cdap-sentry/cdap-sentry-extension/cdap-sentry-binding/src/main/java/co/cask/cdap/security/authorization/sentry/binding/SentryAuthorizer.java]: the "" is so it starts with a `.`. Addressed your other comment.

**Comment by bdmogal** [comment]
http://builds.cask.co/browse/CDAP-CSI35-1

**Comment by rohitsinha54** [comment]
LGTM

**Comment by rohitsinha54** [comment]
Build passed merging.

**Pull request merged**